src/mscorlib/src/System/String.cs

```diff
@@ -1170,6 +1252,35 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
             return foundCount;
         }        
         
+        //--------------------------------------------------------------------
+        // This function returns number of the places within baseString where
+        // instances of the separator string occurs.
+        // Args: separator  -- the separator
+        //       sepList    -- an array of ints for split string indicies.
+        //--------------------------------------------------------------------
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        private unsafe int MakeSeparatorList(string separator, ref int[] sepList) {
+            Contract.Assert(!string.IsNullOrEmpty(separator), "!string.IsNullOrEmpty(separator)");
```

Comment:
String interpolation can't be used as mscorlib still needs to be compiled with the older C# compiler.

As for `String` vs. `string`, the code style in this file is inconsistent all over the place, but `String` would probably be the most consistent with other uses in this file (this is in contrast with the corefx repo where `string` is preferred).